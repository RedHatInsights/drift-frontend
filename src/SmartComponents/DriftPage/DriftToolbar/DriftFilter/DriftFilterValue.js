import React from 'react';
import PropTypes from 'prop-types';
import { ToolbarFilter } from '@patternfly/react-core';
import SearchBar from '../../SearchBar/SearchBar';
import FilterDropDown from '../../FilterDropDown/FilterDropDown';

function DriftFilterValue(props) {
    const { activeFactFilters, factFilter, filterByFact, filterType, handleFactFilter, removeChip, setHistory, stateFilters } = props;

    const setFactFilterChips = () => {
        let factFilterChips = [ ...activeFactFilters ];

        if (factFilter.length && !activeFactFilters.includes(factFilter)) {
            factFilterChips.push(factFilter);
        }

        return factFilterChips;
    };

    const setStateChips = (stateFilters) => {
        let stateChips = [];

        stateFilters.forEach(function(filter) {
            if (filter.selected) {
                stateChips.push(filter.display);
            }
        });

        return stateChips;
    };

    const renderFilterInput = (type) => {
        return <React.Fragment>
            <ToolbarFilter
                chips={ setFactFilterChips() }
                deleteChip={ removeChip }
                deleteChipGroup={ removeChip }
                categoryName="Fact name"
            >
                { type === 'Fact name'
                    ? <SearchBar
                        factFilter={ factFilter }
                        activeFactFilters={ activeFactFilters }
                        handleFactFilter={ handleFactFilter }
                        filterByFact={ filterByFact }
                        setHistory={ setHistory }
                    />
                    : null
                }
            </ToolbarFilter>
            { /*<ToolbarFilter
                chips={ setFactTypeChips() }
                deleteChip={ removeChip }
                deleteChipGroup={ removeChip }
                categoryName="Fact type"
            >
                { type === 'Fact type' ? }
            </ToolbarFilter>*/ }
            <ToolbarFilter
                chips={ setStateChips(stateFilters) }
                deleteChip={ removeChip }
                deleteChipGroup={ removeChip }
                categoryName="State"
            >
                { type === 'State' ? <FilterDropDown setHistory={ setHistory } /> : null }
            </ToolbarFilter>
        </React.Fragment>;
    };

    return (
        <React.Fragment>
            { renderFilterInput(filterType) }
        </React.Fragment>
    );
}

DriftFilterValue.propTypes = {
    activeFactFilters: PropTypes.array,
    factFilter: PropTypes.string,
    filterByFact: PropTypes.func,
    filterType: PropTypes.string,
    handleFactFilter: PropTypes.func,
    removeChip: PropTypes.func,
    setHistory: PropTypes.func,
    stateFilters: PropTypes.array
};

export default DriftFilterValue;
