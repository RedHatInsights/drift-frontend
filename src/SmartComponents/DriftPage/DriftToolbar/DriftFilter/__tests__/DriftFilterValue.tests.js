import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import DriftFilterValue from '../DriftFilterValue';
import stateFilters from '../../../../modules/__tests__/state-filter.fixtures';

describe('DriftFilterValue', () => {
    let props;

    beforeEach(() => {
        props = {
            activeFactFilters: [],
            addStateFilter: jest.fn(),
            clearAllFactFilters: jest.fn(),
            factFilter: '',
            filterByFact: jest.fn(),
            filterType: 'Fact name',
            handleFactFilter: jest.fn(),
            setHistory: jest.fn(),
            stateFilters: stateFilters.allStatesTrue
        };
    });

    it('should render correctly', () => {
        const wrapper = shallow(
            <DriftFilterValue { ...props } />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with fact filter chips', () => {
        props.factFilter = 'dog';
        props.activeFactFilters = [ 'cat', 'mouse' ];

        const wrapper = shallow(
            <DriftFilterValue { ...props } />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render with state filter deselected', () => {
        props.filterType = 'State';
        props.stateFilters[0].selected = false;

        const wrapper = shallow(
            <DriftFilterValue { ...props } />
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });
});
