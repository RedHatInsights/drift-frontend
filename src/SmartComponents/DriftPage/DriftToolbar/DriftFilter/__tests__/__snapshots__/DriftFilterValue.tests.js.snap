// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DriftFilterValue should render correctly 1`] = `
<Fragment>
  <ToolbarFilter
    categoryName="Fact name"
    chips={Array []}
    showToolbarItem={true}
  >
    <SearchBar
      activeFactFilters={Array []}
      factFilter=""
      filterByFact={[MockFunction]}
      handleFactFilter={[MockFunction]}
      setHistory={[MockFunction]}
    />
  </ToolbarFilter>
  <ToolbarFilter
    categoryName="State"
    chips={
      Array [
        "Same",
        "Different",
        "Incomplete data",
      ]
    }
    showToolbarItem={true}
  />
</Fragment>
`;

exports[`DriftFilterValue should render with fact filter chips 1`] = `
<Fragment>
  <ToolbarFilter
    categoryName="Fact name"
    chips={
      Array [
        "cat",
        "mouse",
        "dog",
      ]
    }
    showToolbarItem={true}
  >
    <SearchBar
      activeFactFilters={
        Array [
          "cat",
          "mouse",
        ]
      }
      factFilter="dog"
      filterByFact={[MockFunction]}
      handleFactFilter={[MockFunction]}
      setHistory={[MockFunction]}
    />
  </ToolbarFilter>
  <ToolbarFilter
    categoryName="State"
    chips={
      Array [
        "Same",
        "Different",
        "Incomplete data",
      ]
    }
    showToolbarItem={true}
  />
</Fragment>
`;

exports[`DriftFilterValue should render with state filter deselected 1`] = `
<Fragment>
  <ToolbarFilter
    categoryName="Fact name"
    chips={Array []}
    showToolbarItem={true}
  />
  <ToolbarFilter
    categoryName="State"
    chips={
      Array [
        "Different",
        "Incomplete data",
      ]
    }
    showToolbarItem={true}
  >
    <Connect(FilterDropDown)
      setHistory={[MockFunction]}
    />
  </ToolbarFilter>
</Fragment>
`;
